src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -290,6 +291,14 @@ public function getEnv(string $prefix, string $name, \Closure $getEnv)
             return trim($env);
         }
 
+        if ('not' === $prefix) {
+            if (false === is_bool($env)) {
```

Comment:
I'm not for being strict here.

Reply:
Alright - would [this implementation](https://github.com/symfony/symfony/pull/40169/commits/ce5a996d1f58dc0bae8befbebbb36a4c6b036052) fit better?